state_machine/fsm.go

```diff
@@ -0,0 +1,34 @@
+package state_machine
+
+import (
+	"github.com/looplab/fsm"
+)
+
+// NewNodeFSM returns a KISS Finite State Machine that is meant to mimick the various "states" of the node.
+//
+// source: consensus/doc/PROTOCOL_STATE_SYNC.md + additions for P2P
+// The initial implementation is going to be used to understand, from a P2P perspective, if the node requires bootstrapping
```

Comment:
Not a fan of this as a comment in the codebase - it's very temporal and feels more like a note from a meeting.

If we follow my enum suggestion below, we can add a comment along the lines of:

```
The current set of states and events captures a limited subset of state sync and P2P bootstrapping-related events.
More states & events in any of the modules supported should be added and documented here.
```

Wdyt?

Reply:
Fwiw, a while ago I read an article that said "treat every day as if it's your last day." I follow this methodology in both my work life and my personal life.

Reply:
It was a note for a code-review really :)

And it's working! :P

Reply:
Indeed it is :)